File: RepoTransBench/source_projects/Python/hackebrot_pytest-tricks/public_tests/create_tests_via_parametrization/test_public_parametrize.py
# -*- coding: utf-8 -*-

import pytest

PUBLIC_RULES = (
    (2 * 7, "FooBar"),
    (2, "Foo"),
    (7, "Bar"),
)

def foobar(number):
    for div_number, substitution in PUBLIC_RULES:
        if not number % div_number:
            return substitution
    return str(number)

@pytest.mark.parametrize(
    'number, word', [
        (1, '1'),
        (2, 'Foo'),
        (7, 'Bar'),
        (14, 'FooBar'),
        (8, 'Foo'),
        (13, '13')
    ]
)
def test_foobar(number, word):
    assert foobar(number) == word

File: RepoTransBench/target_projects/Python/Rust/hackebrot_pytest-tricks/tests/public/test_public_parametrize.rs
fn foobar(number: i32) -> String {
    let public_rules = [
        (2 * 7, "FooBar"),
        (2, "Foo"),
        (7, "Bar"),
    ];
    for (div_number, substitution) in public_rules {
        if number % div_number == 0 {
            return substitution.to_string();
        }
    }
    number.to_string()
}

#[test]
fn test_foobar_cases() {
    let test_cases = [
        (1, "1"),
        (2, "Foo"),
        (7, "Bar"),
        (14, "FooBar"),
        (8, "Foo"),
        (13, "13"),
    ];
    for (number, word) in test_cases {
        assert_eq!(foobar(number), word);
    }
}